### ISSUE TITLE
Laravel Passport & Scout

### ISSUE THREAD

**Issue by wouterNL** [apiato/apiato#98]
**TL;DR : 
Are you gonna implement Scout and Passport into Hello-API?**

Let me start off by saying: This code base is AMAZING, great job! 
I have not build a project with it, but I am more exploring it for research purposes for a future project, and using this for it would be amazing. 

But I do have a question about Laravel Passport and Laravel Scout:

Are you planning on implementing these?
I am not too worried about Scout myself, since its pretty straight forward, and I could do that myself.

But the Passport is a whole other story, since this project is so deeply woven with Dingo. 
And Currently Dingo doesn't support Passport, there are some custom providers that kinda work but not really. So what would be a timeline if you plan on adding it? 

Thanks!

Wouter

**Comment by Mahmoudz** [comment]
@wouterNL Thank you.

I do have a plan to remove everything related to Dingo API. But if I'm gonna do it, it's not gonna be very soon!! maybe in a month or two or three!! Because currently I'm building some projects on top of Hello API and I don't have time to mess with it just to avoid errors which can cause delay in my projects.. 

If you or anyone is interested in digging into that and switch it. I'd like to help and answer all questions 24/7 to achieve this as soon as possible.

Best,

**Comment by wouterNL** [comment]
Thanks for your response. 

I am currently looking into implementing passport in my private time. 
But I might be able to also look into completely taking out Dingo since its deprecated with the new functionality of laravel 5.3+. 

So I might be able to contribute when I have the time in the upcoming weeks.

**Comment by wouterNL** [comment]
I have build a laravel passport, scout version without Dingo. 
I have not updated any of the documentation or tested all the endpoints, but I think I am getting pretty close. 

You can check it out here: 
[https://github.com/wouterNL/Hello-API/tree/taking_out_dingo_JWT_replace_with_passport](https://github.com/wouterNL/Hello-API/tree/taking_out_dingo_JWT_replace_with_passport)

If you find any problems please let me know.

**Comment by Mahmoudz** [comment]
@wouterNL thanks, I will check it this weekend, probably tomorrow.

If you could, just make sure to merge all the commits I'm gonna do today with your fork. 

Thanks again,

**Comment by wouterNL** [comment]
I just merged all your latest commits into the branch. 
Just a reminder, It's still a work in progress and you will still find a lot of bugs and old files.

Something to think about: 
Since passport handles almost all authorization methods, is it still necessary to have the Authorization container?

**Comment by Mahmoudz** [comment]
lol I just made another commit now.

Yes for sure it is very important container, used to handle login, logout for api and web, the JWT package is required there so it also contain it's config file, in addition to other stuff like the TokenTrait and some Exceptions... 

I'll wait until you tell me to check the code, so this way we don't mess each other's work. When you feel it's ready, let me know and all the remaining stuff to check it myself. 

If you have any question send me a direct message me on Gitter, anytime.

Best,

**Comment by sajjad-ser** [comment]
I am impatiently waiting for this feature

**Comment by wouterNL** [comment]
I'll put an update on my branch tomorrow! 
There will be some bugs, but minor.

**Comment by wouterNL** [comment]
I present to you: https://github.com/wouterNL/Hello-API/tree/taking_out_dingo_JWT_replace_with_passport

If you have any problems please let me know, and I'll try and fix them as soon as possible. 
I did not test if the WEB routes still work because I do not use that in my project.

**Comment by Mahmoudz** [comment]
@wouterNL looks good, but not sure if it's complete I'm checking branch `taking_out_dingo_JWT_replace_with_passport`. Anyway whenever you're done switching the authentications to passport, please send a PR to test it myself. If your commits already include Scout related stuff it's fine, it might be helpful feature for some people. Best,

**Comment by wouterNL** [comment]
@Mahmoudz  Yeah it's not complete nor has it any tests, but it has all the basic functionality.
Also, I didn't include Scout yet, but I'll do that at a later time. 

I'll send a PR but you might want to check the WEB routes, I deleted those in my project since I didn't need them.
